src/plugins/data/public/search/aggs/test_helpers/mock_agg_types_registry.ts

```diff
@@ -51,10 +52,21 @@ export function mockAggTypesRegistry<T extends BucketAggType<any> | MetricAggTyp
       }
     });
   } else {
-    const aggTypes = getAggTypes({ uiSettings: coreMock.createSetup().uiSettings });
+    const aggTypes = getAggTypes(createMockedAggTypesDependencies());
+
     aggTypes.buckets.forEach(type => registrySetup.registerBucket(type));
     aggTypes.metrics.forEach(type => registrySetup.registerMetric(type));
   }
 
   return registry.start();
 }
+
+/**
+ * Testing utility which creates a mocked instance of AggTypesDependencies,
+ *
+ * @internal
+ */
+export const createMockedAggTypesDependencies = (): AggTypesDependencies => ({
+  core: coreMock.createSetup(),
+  query: queryServiceMock.createSetupContract(),
+});
```

Comment:
If we go the route of only passing in the required deps as I described above, then we wouldn't need a test helper -- each individual agg could import only the mocks they need for their tests.